src/Uno.UI/UI/Xaml/Controls/WebView/UnoWKWebView.iOSmacOS.cs

```diff
@@ -239,6 +291,23 @@ private void OnRunJavaScriptTextInputPanel(WKWebView webview, string prompt, str
 
 			var controller = webview.FindViewController();
 			controller?.PresentViewController(alert, true, null);
+#else
+			var alert = new NSAlert()
+			{
+				AlertStyle = NSAlertStyle.Informational,
+				InformativeText = prompt,
+			};
+			var textField = new NSTextField()
+			{
+				PlaceholderString = defaultText,
+			};
+			alert.AccessoryView = textField;
+			alert.AddButton(OkString);
+			alert.AddButton(CancelString);
+			alert.Buttons[0].Activated += (s, e) => completionHandler(textField.StringValue);
+			alert.Buttons[1].Activated += (s, e) => completionHandler(null);
```

Comment:
When using .Activated, the alert does not close after a button press.
I suggest using `BeginSheetForResponse` instead.

For example:
`
alert.BeginSheetForResponse(webview.Window, (result) => {
	var okButtonClicked = result == 1000;
	completionHandler(okButtonClicked ? textField.StringValue : null);
});
`